packages/scheduler/src/__tests__/SchedulerBrowser-test.js

```diff
@@ -0,0 +1,350 @@
+        'scheduler/src/forks/SchedulerHostConfig.default.js',
+      ),
+    );
+
+    runtime = installMockBrowserRuntime();
+    performance = window.performance;
+    Scheduler = require('scheduler');
+    scheduleCallback = Scheduler.unstable_scheduleCallback;
+    NormalPriority = Scheduler.unstable_NormalPriority;
+  });
+
+  afterEach(() => {
+    if (!runtime.isLogEmpty()) {
+      throw Error('Test exited without clearing log.');
+    }
+  });
+
+  function installMockBrowserRuntime() {
+    let VSYNC_INTERVAL = 33.33;
+
+    let hasPendingMessageEvent = false;
+
+    let rAFCallbackIDCounter = 0;
+    let rAFCallback = null;
+    let isRunningRAFCallback = false;
+
+    let rICCallbackIDCounter = 0;
+    let rICCallback = null;
+
+    let timerIDCounter = 0;
+    // let timerIDs = new Map();
+
+    let eventLog = [];
+
+    const window = {};
+    global.window = window;
+
+    let currentTime = 0;
+
+    window.performance = {
+      now() {
+        return currentTime;
+      },
+    };
+    window.requestAnimationFrame = cb => {
+      if (rAFCallback !== null) {
+        throw Error('rAF already scheduled');
+      }
+      if (isRunningRAFCallback) {
+        log('Request Animation Frame [Reposted]');
+      } else {
+        log('Request Animation Frame');
+      }
+      rAFCallback = cb;
+      return rAFCallbackIDCounter++;
+    };
+    window.cancelAnimationFrame = id => {
+      rAFCallback = null;
+    };
+
+    window.requestIdleCallback = cb => {
+      if (rICCallback !== null) {
+        throw Error('rAF already scheduled');
+      }
+      log('Request Idle Callback');
+      rICCallback = cb;
+      return rICCallbackIDCounter++;
+    };
+    window.cancelIdleCallback = id => {
+      rICCallback = null;
+    };
+
+    window.setTimeout = (cb, delay) => {
+      const id = timerIDCounter++;
+      log(`Set Timer`);
+      // TODO
+      return id;
+    };
+    window.clearTimeout = id => {
+      // TODO
+    };
+
+    const port1 = {};
+    const port2 = {
+      postMessage() {
+        if (hasPendingMessageEvent) {
+          throw Error('Message event already scheduled');
+        }
+        log('Post Message');
+        hasPendingMessageEvent = true;
+      },
+    };
+    global.MessageChannel = function MessageChannel() {
+      this.port1 = port1;
+      this.port2 = port2;
+    };
+
+    function ensureLogIsEmpty() {
+      if (eventLog.length !== 0) {
+        throw Error('Log is not empty. Call assertLog before continuing.');
+      }
+    }
+    function setHardwareFrameRate(fps) {
+      VSYNC_INTERVAL = 1000 / fps;
+    }
+    function advanceTime(ms) {
+      currentTime += ms;
+    }
+    function advanceTimeToNextFrame() {
+      const targetTime =
+        Math.ceil(currentTime / VSYNC_INTERVAL) * VSYNC_INTERVAL;
+      if (targetTime === currentTime) {
+        currentTime += VSYNC_INTERVAL;
+      } else {
+        currentTime = targetTime;
+      }
+    }
+    function getMostRecentFrameNumber() {
```

Comment:
trivial naming nit: `getCurrentFrameNumber`? "Most recent" feels like an odd name to use when `currentTime` is exactly at a frame boundary.

**Edit** I see how you're using this method in the tests below, and `getMostRecentFrameNumber` makes more sense as a name. I think the usage in `fireAnimationFrame` feels more like a "current" value but I don't feel strongly about this. :smile: